modules/mono/glue/GodotSharp/GodotSharp/Core/GodotSynchronizationContext.cs

```diff
@@ -24,5 +24,10 @@ public void ExecutePendingContinuations()
                 workItem.Key(workItem.Value);
             }
         }
+
+        public void Dispose()
+        {
+            _queue.Dispose();
+        }
```

Comment:
Hmm maybe we should dispose this somewhere.

Reply:
Interesting, it's used in `GodotTaskScheduler` but I don't remember getting the same warning about it not being disposable there, I guess it should also be made disposable.

These types seem to be mostly used as singletons (they are static fields) so can't really dispose them. I can dispose `GodotTaskScheduler` in `Dispatcher.InitializeDefaultGodotTaskScheduler` if the static instance wasn't null before assigning although I expect this method to only be called once.